tools/depends/native/TexturePacker/src/decoder/IDecoder.h

```diff
@@ -41,50 +69,30 @@ class DecodedFrame
   DecodedFrame() = default;
   RGBAImage rgbaImage; /* rgbaimage for this frame */
   int delay = 0; /* Frame delay in ms */
+  IDecoder* decoder = nullptr; /* Pointer to decoder */
 };
 
 class DecodedFrames
 {
   public:
     DecodedFrames() = default;
     std::vector<DecodedFrame> frameList;
-    void* user = nullptr; /* used internally*/
-    void (*destroyFN)(void*) = nullptr;
 
     void clear()
     {
       for (auto f : frameList)
       {
-        delete[] f.rgbaImage.pixels;
-      }
-      if (destroyFN)
-      {
-        destroyFN(user);
+        if (f.decoder != NULL)
+        {
+          f.decoder->FreeDecodedFrame(f);
+        }
+        else
+        {
+          fprintf(stderr,
+            "ERROR: %s - can not determine decoder type for frame!\n",
+            __FUNCTION__);
```

Comment:
We normally do not use fprintf and instead call the log, fprintf only if absolutely necessary (e.g. at the start before necessary parts are available).
```cpp
CLog::Log(LOGERROR, "DecodedFrames::{} - can not determine decoder type for frame!", __func__);
```

Reply:
Thanks for approval! I know about CLog, but this utility is not facing an end user and there are no invocations of CLog in `tools/depends/native/TexturePacker/` folder but plenty of `fprintf` calls:

```shell
/tmp/xbmc$ grep -R CLog tools/depends/native/TexturePacker/ | wc -l
0
/tmp/xbmc$ grep -R fprintf tools/depends/native/TexturePacker/ | wc -l
34
```

Reply:
Fair, this is in `depends/native`